airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,179 @@
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from datetime import datetime
+from typing import Optional, Set
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator, poke_mode_only
+from airflow.utils.decorators import apply_defaults
+
+
+@poke_mode_only
+class S3UploadSessionCompleteSensor(BaseSensorOperator):
+    """
+    Checks for changes in the number of objects at prefix in AWS S3
+    bucket and returns True if the inactivity period has passed with no
+    increase in the number of objects. Note, this sensor will no behave correctly
+    in reschedule mode, as the state of the listed objects in the S3 bucket will
```

Comment:
It should throw an error then.

Reply:
It does! The `@poke_mode_only` decorator should do the job!